lib/segment/src/index/struct_payload_index.rs

```diff
@@ -48,6 +49,9 @@ pub struct StructPayloadIndex {
     payload: Arc<AtomicRefCell<PayloadStorageEnum>>,
     /// Used for `has_id` condition and estimating cardinality
     id_tracker: Arc<AtomicRefCell<IdTrackerSS>>,
+    /// Vector storages for each field, used for `has_vector` condition
+    #[allow(dead_code)]
+    vector_storages: HashMap<VectorName, Arc<AtomicRefCell<VectorStorageEnum>>>,
```

Comment:
If the payload index only reads from the vector storage, can't this just be an `Arc<VectorStorageEnum>`?